### DISCUSSION TITLE
.ipynb  contains no metadata

### DISCUSSION THREAD

**Discussion by ommmirage** [quarto-dev/quarto-cli#9987] [Q&A] [answered]
### Description

WARN: File /Users/sergey/DeliciousBlog/posts/welcome/index.ipynb in the listing 'listing' contains no metadata.

https://github.com/ommmirage/ommmirage.github.io/blob/master/DeliciousBlog/posts/welcome/index.ipynb

**Comment by mcanouil** [comment]
If your document contains only plain text, it would be easier (even from a Git perspective) to use a `.qmd` extension.

You can use `quarto convert` to switch between Jupyter Notebook and Quarto document. This would help you see what's wrong in your files.

**Reply by mcanouil** [reply, marked as answer]
This is how the Notebook should look like:

<table>
<tr><th>Preview</th><th>Jupyter Notebook</th></tr>
<tr><td>

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/dc17e322-49b8-4dc8-aa8c-cc2ebaff4475">

</td><td>

````ipynb
{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "---\n",
        "title: Welcome to my blog!\n",
        "date: 2024-06-13\n",
        "format: html\n",
        "jupyter: python3\n",
        "---\n",
        "\n",
        "Hello, everyone!\n",
        "\n",
        "My name is Sergey, I'm from Astrakhan, Russia.\n",
        "\n",
        "Last week, I started my journey into Deep Learning with the amazing teacher Jeremy Howard. His method fits me perfectly!\n",
        "\n",
        "* He is one of the best experts in the field\n",
        "* His course is modern\n",
        "* He teaches through practice\n",
        "* He simplifies as much as possible\n",
        "* I'm supposed to create a model and improve it during the course\n",
        "* We should dig deeper and deeper\n",
        "\n",
        "To succeed in DL (Deep Learning) you don't need academic background, but you need to apply what you learn from the course to your personal project and you should always be persistent.\n",
        "\n",
        "By the end of the course you'll understand all the code that's inside fast.ai, which means you've learnt the most important best practicies used in modern DL.\n",
        "\n",
        "This is my motivation, which I wrote down on my flipchart. Btw, try to write your ideas on the flipchart, maybe you'll like it as much as I do) \n",
        "\n",
        "Although Jeremy says that I should create my project right now, I'm not really interested in computer vision, which we've learned about during the first two lessons. I'm more interested in natural language processing, which we'll learn in lesson 3, and video processing.\n",
        "\n",
        "Here is the [link to his course](https://course.fast.ai/).\n",
        "\n",
        "I also have a [telegram chanel](http://https://t.me/cash900) on russian."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {},
      "source": [
        "2+2"
      ],
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 4
}
````

</td></tr>
</table>

I don't know how you created your Notebook but I suggest to follow the tutorial to understand what is expected by Quarto: https://quarto.org/docs/get-started/hello/jupyter.html

**Comment by cderv** [comment]
> WARN: File /Users/sergey/DeliciousBlog/posts/welcome/index.ipynb in the listing 'listing' contains no metadata.

This is a warning from Quarto that tells you that your listing with id `listing` has no metadata. I recognized this is quite generic but here is why you get, which you could probably have guess with our doc helps; 

* You ipynb is part of `posts` folder that you asked to in your `_quarto.yml` to be a listing (https://github.com/ommmirage/ommmirage.github.io/blob/cdabfa70bd4758ea9794d69963d6790048069fef/DeliciousBlog/index.qmd#L3-L9) 
* This ipynb does not have any Quarto metadata (i.e. YAML block at the top). See how get started how to use a raw or markdown cell to include this (https://quarto.org/docs/get-started/hello/jupyter.html) 
* Listings does expect some metadata to be set in each document so that they can be shown in the listing page. [Listing fields](https://quarto.org/docs/get-started/hello/jupyter.html) in the listing page (in table form for example) come from each document. 

So you just need to follow what the warning is trying to tell you: You have a listing in your project, which needs to include a file that have no metadata. We don't error, but we warn because this is not expected. 

Hope it helps understand
